packages/grafana-data/src/types/datasource.ts

```diff
@@ -370,6 +370,12 @@ abstract class DataSourceApi<
    * @alpha -- experimental
    */
   getDefaultQuery?(app: CoreApp): Partial<TQuery>;
+
+  /**
+   * Optionally, use this method to render details for the health check
+   * Don't forget to change the file extension to "tsx" to enable webpack's JSX transformer
+   */
+  renderHealthCheckDetails?(details: unknown): JSX.Element;
```

Comment:
Instead of allowing the plugin developers to implement a `renderHealthCheckDetails` function. What do you think about giving them the possibility to register a HealthCheckDetails component in `module.ts(x)` that we use to render this.

Even if it is possible to supply `jsx` via the `DataSourceApi` it feels a bitt odd.

Reply:
Great suggestion! I'll be going in that direction – will probably be looking for some feedback as I'm not too familiar with the plugin architecture :)